pulsar-jms/src/main/java/com/datastax/oss/pulsar/jms/PulsarConnectionConsumer.java

```diff
@@ -90,7 +94,18 @@ public void close() throws JMSException {
       return;
     }
     try {
-      this.spool.join();
+      if (closeTimeout > 0) {
+        this.spool.join(closeTimeout);
+        if (this.spool.isAlive()) {
+          log.warn(
+              "Couldn't stop the ConnectionConsumer on {} within "
+                  + "the configured jms.connectionConsumerStopTimeout={}",
+              this.consumer.getDestination(),
+              closeTimeout);
+        }
```

Comment:
Would it make sense to call `spool.interrupt()` as the last resort?

Reply:
the user case is that the process is going to die. Interrupting a thread may have unpredictable behaviour if the code does not handle InterruptedException or the interrupted state.

btw I have applied your suggestion